deepeval/dataset/dataset.py

```diff
@@ -404,58 +405,179 @@ def pull(self, alias: str, auto_convert_goldens_to_test_cases: bool = True):
+                subject=subject,
+                task=task,
+                output_format=output_format,
+                num_initial_goldens=num_initial_goldens,
+                num_evolutions=num_evolutions,
+                enable_breadth_evolve=enable_breadth_evolve,
+                _show_indicator=_show_indicator,
+                evolution_types=evolution_types
+            )
+        )
+        
+    
+    def generate_goldens_from_prompts(
+        self,
+        prompts: List[str],
+        num_evolutions: int = 1,
+        enable_breadth_evolve: bool = False,
+        _show_indicator: bool = True,
+        evolution_types: List[PromptEvolution] = [
+            PromptEvolution.REASONING,
+            PromptEvolution.CONCRETIZING,
+            PromptEvolution.CONSTRAINED,
+            PromptEvolution.COMPARATIVE,
+            PromptEvolution.HYPOTHETICAL,
+        ],
+    ):
+        from deepeval.synthesizer import Synthesizer
+        synthesizer = Synthesizer()
+
+        self.goldens.extend(
+            synthesizer.generate_goldens_from_prompts(
+                prompts=prompts,
+                num_evolutions=num_evolutions,
+                enable_breadth_evolve=enable_breadth_evolve,
+                _show_indicator=_show_indicator,
+                evolution_types=evolution_types,
+            )
+        )
+        
+        
+    def generate_red_team_goldens(
+        self,
+        contexts: List[List[str]],
+        include_expected_output: bool = False,
+        max_goldens_per_context: int = 2,
+        num_evolutions: int = 3,
+        enable_breadth_evolve: bool = False,
+        evolution_types: List[RedTeamEvolution] = [
+            RedTeamEvolution.PROMPT_INJECTION,
+            RedTeamEvolution.PROMPT_PROBING, 
+            RedTeamEvolution.GRAY_BOX_ATTACK, 
+            RedTeamEvolution.JAIL_BREAKING
+        ],
+        responses: List[Response] = [
+            Response.BIAS,
+            Response.DATA_LEAKAGE,
+            Response.HALLUCINATION,
+            Response.OFFENSIVE,
+            Response.UNFORMATTED
+        ],
+        use_case: UseCase = UseCase.QA,
+        _show_indicator: bool = True,
+    ):
+        from deepeval.synthesizer import Synthesizer
+        synthesizer = Synthesizer()
+
+        self.goldens.extend(
+            synthesizer.generate_red_team_goldens(
+                contexts=contexts,
+                include_expected_output=include_expected_output,
+                max_goldens_per_context=max_goldens_per_context,
+                num_evolutions=num_evolutions,
+                enable_breadth_evolve=enable_breadth_evolve,
+                evolution_types=evolution_types,
+                _show_indicator=_show_indicator,
+                responses=responses,
+                use_case=use_case,
+            )
+        )
 
     def generate_goldens(
         self,
         contexts: List[List[str]],
-        synthesizer: BaseSynthesizer = None,
```

Comment:
Was this deleted because of circular import? Need to let users define their own synthesizer, can make it untyped `synthesizer = None` but check for type in the method using `is_instance(synthesizer, BaseSynthesizer)`